### DISCUSSION TITLE
Notifications for KNX and system events

### DISCUSSION THREAD

**Discussion by brutella** [brutella/hkknx-public#265] [Ideas]
HomeKit already supports basic notifications, for example notify me when the motion sensor detects motion at night or when a door lock is opened. Those notifications can be managed in the Apple Home app.

hkknx also supports notifications in scripts using the function [hkknx.SendNotification](https://hochgatterer.me/pkg/hkknx/#SendNotification). Scripts are much more flexible and can send notifications for all kinds of events, for example when it gets windy in the upcoming night, you can send a notification to your iPhone.

All of this is great but has it's limitations.

- HomeKit notifications are predefined and can only be customised to a certain extend.
- HomeKit notifications only work for specific accessories. You cannot setup a notification when a light is turned on.
- hkknx notifications are currently only available in scripts; basic programming skills are required.

---

My idea would be to make notifications accessible via the web ui and use them for different events.

**Accessory events**
You can use any characteristics for any accessory type to trigger a notification.     

- Send a notification to only my iPhone when the the door is still unlocked at 10PM.
- Send a notification when a specific light is turned on at night.

**KNX events**
You can trigger notifications if a value in a specific value range is received from the KNX bus. 

- Send a notification if the value 1 is received from the group address 1/2/3.
- Send a notification if a value between 0 and 50% is received from the group address 1/2/3 but only between 9PM and 6AM.

**System events**
Send a notification if specific system events happen.

- Send me a notification if the connection to the KNX gateway is closed.
- Send me a notification if the CPU utilisation is higher than 80% for more than 1 minute.
- Send me a notification if the disk space is higher than 80%.

What do other people think of this idea? 
Is this feature actually needed and are there other use cases?

**Comment by chef76** [comment]
Great Feature!

I would prefer to configure this Notifications in the Configurator App, like it is in Home. This would be great for normal Users who don't use the Web-Interface because they get it ready-to-use configured.

**Reply by brutella** [reply]
That's a great idea!

**Comment by petnau** [comment]
Yes please!

**Comment by kevinkandziora** [comment]
Hey @brutella,

I think this is a great idea and would be very helpful. Maybe as a friendly intention, but is it possible to read states of non-KNX device like additional accessoires, like e.g. eve window & door contact sensors? Would be cool, if non-KNX and KNX-devices could work closer with each other.

**Comment by Roughguess** [comment]
Hello brutella, I have a binary contact connected to my doorbell and also receive a notification via the dot sensor via hkknx. Can I also play a sound via my HomePods with this?

**Reply by brutella** [reply]
I think it should be possible to play a sound via the HomePod when the doorbell is pressed by creating an automation in the Apple Home app. The trigger would be the press of the doorbell, and the action is letting the HomePod play a sound.
